### PR TITLE
added sklearn GLM

### LINKED ISSUE #214: Support scikit-learn GLM models
Wow, the newest scikit-learn release [has introduced some GLM models](https://scikit-learn.org/stable/auto_examples/release_highlights/plot_release_highlights_0_23_0.html#generalized-linear-models-and-poisson-loss-for-gradient-boosting)!

Should be good competitor for [recently added GLMs from `statsmodels`](https://github.com/BayesWitnesses/m2cgen/pull/195).

https://scikit-learn.org/stable/modules/classes.html#generalized-linear-models-glm-for-regression
- [ ] `linear_model.PoissonRegressor`
- [ ] `linear_model.TweedieRegressor`
- [ ] `linear_model.GammaRegressor`

### PR DESCRIPTION
**Pull request by StrikerRUS**
Closed #214.

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Supported models
-scikit-learn==0.22.2.post1
+scikit-learn==0.23.0
```
Comment (StrikerRUS) [inline requirements-test.txt:2]: I think that as scikit-learn doesn't provide Python 3.5 wheels anymore and Python 3.5 blocks us from better performance (#209) we can drop Python 3.5 support right now.
https://pypi.org/project/scikit-learn/#files

File: requirements-test.txt [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 xgboost==1.0.2
 lightgbm==2.3.1
 statsmodels==0.11.1
-git+git://github.com/scikit-learn-contrib/lightning.git@b96f9c674968496e854078163c8814049a7b9f43
+git+git://github.com/scikit-learn-contrib/lightning@refs/pull/142/head
```
Comment (StrikerRUS) [inline requirements-test.txt:6]: https://github.com/scikit-learn-contrib/lightning/pull/142

I don't believe that that my PR will be merged (at least soon), but this trick allows us to install the latest scikit-learn with lightning at Travis and run all our tests. At the same time, users are still able to install latest `master` of lightning with old scikit-learn (0.23.0) and not face any problems with a such environment setup.

File: tests/e2e/test_e2e.py [hunk 3]
```diff
@@ -275,13 +276,19 @@ def regression_bounded(model, test_fraction=0.02):
...
             random_state=RANDOM_SEED)),
         regression(linear_model.Ridge(random_state=RANDOM_SEED)),
         regression(linear_model.RidgeCV()),
         regression(linear_model.SGDRegressor(random_state=RANDOM_SEED)),
         regression(linear_model.TheilSenRegressor(random_state=RANDOM_SEED)),
+        regression(linear_model.TweedieRegressor(power=0.0)),
+        regression(linear_model.TweedieRegressor(power=1.0)),
+        regression(linear_model.TweedieRegressor(power=1.5)),
+        regression(linear_model.TweedieRegressor(power=2.0)),
+        regression(linear_model.TweedieRegressor(power=3.0)),
```
Comment (StrikerRUS) [inline tests/e2e/test_e2e.py:291]: Refer to https://github.com/scikit-learn/scikit-learn/blob/483cd3eaa3c636a57ebb0dc4765531183b274df0/sklearn/linear_model/_glm/glm.py#L525-L543.

File: requirements-test.txt (hunk 1 above)
Comment (StrikerRUS) [reply requirements-test.txt]: Another one reason to drop support of Python 3.5:
```
/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/xgboost/__init__.py:28: FutureWarning: Python 3.5 support is deprecated; XGBoost will require Python 3.6+ in the near future. Consider upgrading to Python 3.6+.
  FutureWarning)
```

File: requirements-test.txt (hunk 2 above)
Comment (StrikerRUS) [reply requirements-test.txt]: Wow! They merged it! 🎉
